src/index.ts

```diff
@@ -112,31 +113,53 @@ export class SilentPayment {
   }
 
   /**
-   * Serializes a 32-bit unsigned integer i as a 4-byte little-endian
-   */
+   * Serializes a 32-bit unsigned integer i as a 4-byte big-endian
+   * @param i {number} The number to serialize
+   * @returns {Buffer} The serialized number
+   * @private
+   * */
   static _ser32(i: number): Buffer {
     const returnValue = Buffer.allocUnsafe(4);
     returnValue.writeUInt32BE(i);
     return returnValue;
   }
 
+  /**
+   * Sums the private keys of the UTXOs
+   * @param utxos {UTXO[]}
+   * @returns {Buffer} The sum of the private keys
+   * @private
+   **/
   private static _sumPrivkeys(utxos: UTXO[]): Buffer {
-    let ret = ECPair.fromWIF(utxos[0].WIF).privateKey;
-
-    // If taproot, check if the seckey results in an odd y-value and negate if so
-    if (utxos[0].is_taproot && ecc.pointFromScalar(ret)![0] === 0x03) {
-      ret = Buffer.from(ecc.privateNegate(ret));
+    if (utxos.length === 0) {
+      throw new Error("No UTXOs provided");
     }
-    for (let c = 1; c < utxos.length; c++) {
-      let negated_key = ECPair.fromWIF(utxos[c].WIF).privateKey;
+
+    const keys: Array<Buffer> = []
+    for (const utxo of utxos) {
+      let key = ECPair.fromWIF(utxo.WIF).privateKey;
+
+      if (key === undefined) {
+        throw new Error("No private key found for UTXO");
```

Comment:
not found? maybe, invalid?
i think `ECPair.fromWIF()` will throw if you feed undefined or garbage to it. not sure

Reply:
I've added this check, because according to ecpair typescript types ECPair.fromWIF() returns ECPairInterface with optinal `privateKey`. So I just added this check to make typescript happy, maybe it's redundant.